### PR TITLE
[hive] Parse calcite to convert Chinese comments into unicode values whose encoding is UTF-16LE

### LINKED ISSUE #1262: [Bug] [hive] Field Chinese comments cannot be displayed normally when using hive to query table metadata
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Paimon version

0.5-SNAPSHOT

### Compute Engine

hive

### Minimal reproduce step

1. Flink SQL uses hive metastore to create catalog.

2.Flink SQL creates a table with Chinese comments.

3.Hive SQL query this table `desc formatted table_name`

### What doesn't meet your expectations?

Field Chinese comments cannot be displayed normally when using hive to query table metadata

image

image

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by zhuangchong**
…

### Purpose

fix #1262 

### Tests

iShot_2023-05-30_19 03 51

paimon schema:

image

flink sql query;

image

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java [hunk 1]
```diff
@@ -575,7 +582,26 @@ public static Schema fromCatalogTable(CatalogTable catalogTable) {
...
                 .filter(c -> c.getComment().isPresent())
-                .collect(Collectors.toMap(UnresolvedColumn::getName, c -> c.getComment().get()));
+                .collect(
+                        Collectors.toMap(
+                                UnresolvedColumn::getName,
+                                c -> {
+                                    String comment = c.getComment().get();
+                                    if (comment.toUpperCase().startsWith("U&'")) {
+                                        try {
+                                            SqlParser parser = SqlParser.create(comment + "'");
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:592]: I am more worried about whether there is another way? The introduction of the calcite dependency looks a bit heavy, is it possible to just simply deal with it?

Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:592]: We need test to cover chinese comments, for Flink, for Hive

Comment (wxplovecc) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:592]: we use ``update_time` 	 timestamp 	 COMMENT _utf8'更新时间'` this type to define comment and replace `_utf8`

Comment (zhuangchong) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:592]: @wxplovecc This method is good, but we can't force all users to write like this `... COMMENT _utf8'更新时间'`

Comment (wxplovecc) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java:592]: Right

Comment (zhuangchong) [reply paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/FlinkCatalog.java]: https://github.com/apache/incubator-paimon/blob/fbff232ac43c6e19079c0d0c373601a3a54871f2/paimon-hive/paimon-hive-connector-common/src/test/java/org/apache/paimon/hive/HiveCatalogITCaseBase.java#L247-L263

**Comment by JingsongLi** [comment]
@wxplovecc raise a very good point, the memories I once had came back to me... I'm not sure if we have side effects from converting in `FlinkCatalog` like this. But `_utf8'更新时间'` looks a good solve method to this problem.

**Comment by zhuangchong** [comment]
> @wxplovecc raise a very good point, the memories I once had came back to me... I'm not sure if we have side effects from converting in `FlinkCatalog` like this. But `_utf8'更新时间'` has problem too, "utf8" prefix...

I didn't think of a very good solution. At present, this solution has to judge whether to start with `U&'` to convert to minimize the impact.

**Comment by JingsongLi** [comment]
involving @lincoln-lil , this seems a Flink bug.
We can fix it in paimon, but should be consistent with flink.

**Comment by zhuangchong** [comment]
> 

I understand that this is not a bug of flink, because other engines (hive, spark...)do not support the analysis of calcite's UTF-16LE unicode

**Comment by zhuangchong** [comment]
> involving @lincoln-lil , this seems a Flink bug. We can fix it in paimon, but should be consistent with flink.


I understand that this is not a bug of flink, because other engines (hive, spark...)do not support the analysis of calcite's UTF-16LE unicode.

**Comment by lincoln-lil** [comment]
@JingsongLi thanks for pointing it out! I'll investigate it and try to get the solution.

**Comment by lincoln-lil** [comment]
At first glance, when Flink interacts with CatalogTable, it directly passes the Calcite's NlsString comment as a string to the comment attribute of the schema and column. In theory, a Java string should be passed here. I will spend some time verifying the specific impact of the changes in Flink.

**Comment by lincoln-lil** [comment]
> #1262
flink issue created https://issues.apache.org/jira/browse/FLINK-32249

**Comment by zhuangchong** [comment]
Thanks @JingsongLi @lincoln-lil 

I will close this PR and wait for Flink to fix this issue.

**Pull request closed without merging**